tools/integration_tests_runner.py

```diff
@@ -1,167 +1,232 @@
 from __future__ import print_function
 
 import logging
+from multiprocessing import cpu_count
 import multiprocessing
 from optparse import OptionParser
 from os import path
-from Queue import Queue
-from random import shuffle
 import shutil
 import subprocess
 import tempfile
-from threading import Lock
-from threading import Thread
-from time import time
-import traceback
-
 
 from run_desktop_tests import tests_on_disk
 
 __author__ = 't.danshin'
 
-
+TEST_RESULT_LOG = "test_result_log"
 TEMPFOLDER_TESTS = ["search_integration_tests", "storage_integration_tests"]
 
+FILTER_KEY = "--filter"
+DATA_PATH_KEY = "--data_path"
+RESOURCE_PATH_KEY = "--user_resource_path"
+
+
+def setup_test_result_log(log_file, level=logging.INFO):
+    logger = logging.getLogger(TEST_RESULT_LOG)
+    formatter = logging.Formatter("BEGIN: %(file)s\n%(message)s\nEND: %(file)s | result: %(result)d\n")
+    file_handler = logging.FileHandler(log_file, mode='w')
+    file_handler.setFormatter(formatter)
+    logger.setLevel(level)
+    logger.propagate = False
+    logger.addHandler(file_handler)
+
+
+def setup_jenkins_console_logger(level=logging.INFO):
+    formatter = logging.Formatter('%(process)s: %(msg)s')
+    # Time is logged by Jenkins. Any log files on disk will be removed by
+    # Jenkins when the next job starts
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+    multiprocessing.get_logger().setLevel(level)
+    multiprocessing.get_logger().addHandler(stream_handler)
+
+
+def log_start_finish(func):
+    def func_wrapper(test, keys):
+        logger = multiprocessing.get_logger()
+        logger.info("start: >{0} {1}".format(test, keys))
+        result = func(test, keys)
+        logger.info("end: >{0} {1}".format(test, keys))
+        return result
+
+    return func_wrapper
+
+
+@log_start_finish
+def execute_file(test, keys):
+    spell = ["{test} {keys}".format(test=test, keys=keys)]
```

Comment:
Looks like the method should be renamed to execute_test, right? As it takes test and keys to it.